<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="description" content="Erds-exome.GitHub.com : ERDS-exome: a hybrid approach for copy number variants detection from whole-exome sequencing data">

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>ERDS-exome</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/coderChase/jxcnv">View on GitHub</a>

          <h1 id="project_title">ERDS-exome</h1>
          <h2 id="project_tagline">A hybrid approach for copy number variants detection from whole-exome sequencing data</h2>

          <div id="downloads">
            <div class="tar_download_link" onclick="document.location='https://erds-exome.github.io/erds-exome.tar.gz';">
            <a href="erds-exome.tar.gz">werwerwe</a>
            </div>
            <div class="zip_download_link" onclick="document.location='https://erds-exome.github.io/erds-exome.zip';">
            <a href="erds-exome.zip">werwerw</a>
            </div>
          </div>
        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
      	<div id="inner_left">
      		<div>
      			<a href="index.html">Home</a>
      		</div>
      		
      		<div>
      			<a href="tutorial.html">Tutorial</a>
      		</div>
      		<div>
      			<a href="quick_start.html">Quick Start</a>
      		</div>
      		<div>
      			<a href="download.html">Download</a>
      		</div>
      		<div>
      			<a href="contact.html">Contact</a>
      		</div>
      	</div>
      	<div id="inner_right">
		    <h3>
				<a id="erds-exome" class="anchor" href="#erds-exome" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
				Turorial
				</h3>
        <p>
         This is a general tutorial for using ERDS-exome. You can also try ERDS-exome simply using test data. Please read the <a href="quick_start.html">Quick Start Guide</a>.
        </p>
        <p>
          <h4>
            General command structure
          </h4>
          ERDS-exome is a command-line python program (erds_exome.py). There are several sub-commands, each with a set of command line options. In general, the ERDS-exome pipeline consists of the following steps: 
        </p>
        <p><strong>1. Calculate RPKM values for all samples.</strong></p>
<pre>
  python erds_exome.py rpkm [...]
</pre>
        <p><strong>2. Merge RPKM values of all samples into a matrix.</strong></p>
<pre>
  python erds_exome.py merge_rpkm [...]
</pre>
        <p><strong>3. Filter RPKM matrix by filter parameters.</strong></p>
<pre>
  python erds_exome.py filter [...]
</pre>
        <p><strong>4. Normalization.</strong></p>
        <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<strong>In mode 1:</strong> Single-sample based normalization, normalize RPKM matrix by mappability, GC content and exon length.</p>
<pre>
  python erds_exome.py norm_rpkm [...]
</pre>
        <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<strong>In mode 2:</strong> Pooled-sample based normalization, normalize RPKM matrix by singular value decomposition (SVD).</p>
<pre>
  python erds_exome.py svd [...]
</pre>
        <p><strong>5. CNV calling.</strong></p>
        <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<strong>In mode 1:</strong> CNV calling by single-sample mode.</p>
        <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<strong>In mode 2:</strong> CNV calling by pooled-sample mode.</p>  
<pre>
  python erds_exome.py discover [...]
</pre>
        <p><strong>6. Merge CNV calling results.</strong></p>
<pre>
  python erds_exome.py merge [...]
</pre>

    <blockquote>
      <h4>0. Requirements and Setup</h4>
      ERDS-exome requires Python v2.7 (Python 3 is not yet supported). And <a href="http://www.numpy.org/">NumPy</a> package should be installed. If you want to calculate RPKM values from BAM files, <a href="https://github.com/pysam-developers/pysam">pysam</a> is required. Pysam is also used to calculate GC content. In addition, <a href="https://pypi.python.org/pypi/bx-python/0.7.3">bx-python</a> is required to calculate mappability, and <a href="https://github.com/jamescasbon/PyVCF">PyVCF</a> is required to generate SNV heterozygosity information and tagSNP linkage disequilibrium (LD) information.
    </blockquote>

    <blockquote>
      <h4>1. Create probe/target definition file</h4>
      This file defines the chr:start-stop coordinates for the probes/targets in the exome capture. It should be tab-delimited with the following header and columns:
    
<pre>
chr&#9;start&#9;stop
1&#9;14642&#9;14882
1&#9;14943&#9;15063
1&#9;15751&#9;15990
1&#9;16599&#9;16719
1&#9;16834&#9;17074
</pre>
Where the first column represents the number of chromsome, and the second and third columns refer to the start index and end index of target respectively. ERDS-exome also support target file of <strong>bed</strong> format.
</blockquote>

    <blockquote>
      <h4>2. Create RPKM files</h4>
      Using pysam package, ERDS-exome can calculate RPKM from aligned and indexed BAM files.
<pre>
  python erds_exome.py rpkm \
  --input $bamlist_file \
  --maq $map_quality
  --target $target_file \
  --output $output_path
</pre>
			Where $target_file is the probes/target created in former step, $bamlist_file includes the bam files of all samples, and $map_quality is the minimum mapping quality score of reads. The format of $bamlist is like:
<pre>
  NA06984	/data/rjtan/1000GP/exome/NA06984.mapped.ILLUMINA.bwa.CEU.exome.20120522.bam	CEU
  NA06985	/data/rjtan/1000GP/exome/NA06985.mapped.ILLUMINA.bwa.CEU.exome.20130415.bam	CEU
  NA06994	/data/rjtan/1000GP/exome/NA06994.mapped.ILLUMINA.bwa.CEU.exome.20120522.bam	CEU
  NA07000	/data/rjtan/1000GP/exome/NA07000.mapped.ILLUMINA.bwa.CEU.exome.20130415.bam	CEU
</pre>
Where the first column refers to sample name, the second column refers to directory and name of bam file, and the third column refers to ethnic group of the corresponding sample.
    </blockquote>

<blockquote>
      <h4>3. Merge RPKM files</h4>
      Merge the RPKM files from different samples into a matirx.
<pre>
  python erds_exome.py merge_rpkm \
	--rpkm_dir $rpkm_files \
	--target $target_file
</pre>
			Where $rpkm_files includes all RPKM files. There should not be other files except RPKM files in $rpkm_files.
 </blockquote>

 <blockquote>
      <h4>4. Filter RPKM matrix</h4>
      Filter the RPKM matrix by mappability, 
<pre>
  python erds_exome.py filter \
  --rpkm_matrix $RC_matrix.raw \
  --ref_file $ref \
  --map_file $mapability_file \
  --filter_params $filter_params_file
</pre>
Where $RC_matrix.raw is the targets by samples matrix generated in former step. $ref is human reference file (hg19) which can be downloaded <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/">here</a> and $map_file refers to mappability file which can be downloaded <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign100mer.bigWig">here</a>. $filter_params_file contains the filter parameters. Specifically, the format of filter parameter file is like:
<pre>
  23	80	2	5000	0.001
</pre>
The five parameters refer to minimum GC content, maximum GC content, minimum mappability value, maximum exon length and minimum RPKM value respectively.
 </blockquote>

<blockquote>
      <h4>5. Normalization</h4>
      <strong>In mode 1:</strong> Normalize RPKM matrix by three steps, including mappability, GC content and exon length.
<pre>
  python erds_exome.py norm_rpkm \
  --rpkm_matrix $RPKM_matrix.raw.filtered
</pre>
			<strong>In mode 2:</strong> Normalize RPKM matrix by SVD.
<pre>
  python erds_exome.py svd \
  --rpkm_matrix $RPKM_matrix.raw.filtered
</pre>
 </blockquote>

<blockquote>
      <h4>6. CNV calling</h4>
	<strong>In mode 1:</strong> CNV calling by single sample mode. User can specify the HMM parameters file by '--params'. If '--hetsnp' is set to 'True', then the SNV heterozygosity information will be included. If '--tagsnp' is set to 'True', then the tagSNP linkage disequilibrium (LD) information will be included and a tagSNP file should be specified by '--tagsnp_file'. '--hetsnp' and '--tagsnp' are optional parameters. The default options of them are 'False'. If user specify a sample name by '--sample', ERDS-exome will only call CNV for this sample. Otherwise ERDS-exome will call CNV for all samples. 
<pre>
  python erds_exome.py discover \
  --params params.txt \
  --rpkm_matrix $RPKM_matrix.raw.filtered.normalized \
  --mode single \
  --sample $sample_name \
  --vcf $snv_vcf_file \
  --hetsnp True \
  --tagsnp True \
  --tagsnp_file $tagsnp_file \
  --output $sample_name.single.Het.Tag.cnv
</pre>
      <strong>In mode 2:</strong> CNV calling by pooled sample mode. All parameters are the same with CNV calling by single sample mode, except that '--mode' is set to 'pooled'.
<pre>
  python erds_exome.py discover \
  --params params.txt \
  --rpkm_matrix $RPKM_matrix.raw.filtered.SVD \
  --mode pooled \
  --sample $sample_name \
  --vcf=$snv_vcf_file \
  --hetsnp True \
  --tagsnp True \
  --tagsnp_file=$tagsnp_file \
  --output $sample_name.pooled.Het.Tag.cnv
</pre>
 </blockquote>


<blockquote>
      <h4>7. Merge CNV calling results</h4>
			Merge CNV calling results from different methods.      
<pre>
  python erds_exome.py merge \
  --datafile_svd $sample_name.pooled.Het.Tag.cnv \
  --datafile_dis $sample_name.single.Het.Tag.cnv \
  --output $sample_name.merged.Het.Tag.cnv
</pre>
 </blockquote>


		</div>
      </section>
    </div>





    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
